tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -54,11 +54,20 @@
 def matching_types(pyccel_result, python_result):
     """  Returns True if the types match, False otherwise
     """
-    if isinstance(python_result, np.generic):
-        #TODO: Remove when #735 is fixed
-        return isinstance(pyccel_result, (type(python_result.item()), type(python_result)))
+    if isinstance(pyccel_result, np.generic):
+        python_type_alignment = np.dtype(type(pyccel_result.item())).alignment
+        returned_type_alignment = np.dtype(type(pyccel_result)).alignment
+        if python_type_alignment != returned_type_alignment:
+            return isinstance(pyccel_result, type(python_result))
+        else:
+            return isinstance(pyccel_result, type(python_result)) or \
+                    isinstance(pyccel_result.item(), type(python_result))
     else:
-        return isinstance(pyccel_result, type(python_result))
+        if isinstance(python_result, np.generic):
+            #TODO: Remove when #735 is fixed
+            return isinstance(pyccel_result, (type(python_result.item()), type(python_result)))
+        else:
+            return isinstance(pyccel_result, type(python_result))
 
```

Comment:
It seems to me that when `isinstance(pyccel_result, type(python_result)) == True` this function always returns `True`. If that is correct, we can greatly simplify this code with an early exit condition:

```python
if type(pyccel_result) is type(python_result):
    return True
```

And then treat the other corner cases.

Reply:
You are right. Done in ac9de23